src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -109,7 +109,7 @@ static void setup_track_designer_objects()
  *
  *  rct2: 0x006AA82B
  */
-void setup_in_use_selection_flags()
+void setup_in_use_selection_flags(bool onlyUsed = false)
```

Comment:
The name of the argument is not very descriptive. What does it mean when this is true or false?

Also personally I'm not a fan of default argument values, but that's just a preference for explicitly.

Reply:
The idea was that when it's true, it only sets the selection flags of items that are used, specifically only the flags indicating that it's used and selected. If it's false, it'll just get all the flags, for all objects. 
The purpose of this fix is to prevent the situation where the user deselects something, then opens the scenery window and places that object, to then close the object selection. Normally it would look at the flags, notice it's not selected, and thus unload the object, causing corruption.
So here it's doing a check to again set all used scenery checked to prevent if from being unloaded. However, without this bool check, it would be impossible to unselect anything, whether it is used or not. 
I'm open to a better name though.

Reply:
> 

Would you rather have a different name?

Reply:
It might look a bit easier to read if you used an enum class and defined the two states.
```cpp
enum class ObjectSelectionCheck : bool
{
    all = false,
    // When closing the object selection window we want to perform a check to ensure we are not unloading in use objects. Skips all other checks.
    onlyUsed = true, 
}
```

Reply:
Alright, I'll change that, but I am kinda at a loss of how. This was my first PR, and I foolishly hadn't made a branch, and worked on my develop clone. Since these commits kept appearing on any branch I made later, I reset my develop to be identical to the one here. (though I did make a backup develop branch where these changes still exist). But how do I get this PR to start pulling from that branch now? Might it be easier to just make a new PR and redo the changes, now in a proper branch?

Reply:
Well this is Gymnasiast's branch, [Gymnasiast:backport-mm-16486](https://github.com/Gymnasiast/OpenRCT2/tree/backport-mm-16486), which I don't think you have write access to. One of us from the team will have to do that.